fastlane/lib/fastlane/runner.rb

```diff
@@ -149,6 +153,12 @@ def execute_action(method_sym, class_ref, arguments, custom_dir: '..')
       end
     end
 
+    def execute_flow_block(block, current_platform, lane, parameters)
+      # Call the platform specific block and then the general one
+      block[current_platform].call(lane, parameters) if block[current_platform] && current_platform
+      block[nil].call(lane, parameters) if block[nil]
```

Comment:
It seems like it's a bit messy as it is now (this has been here before, so nothing that was introduced with this PR). I'd love to take this opportunity to improve this and clean it up 👍